store/views.py

```diff
@@ -1,82 +1,263 @@
+import datetime
+from django.http import Http404
 from django.shortcuts import render
 from django.shortcuts import get_object_or_404
 from store.models import *
 from django.contrib.auth.decorators import login_required
 from django.http import JsonResponse
 from django.views.decorators.csrf import csrf_exempt
 
-# Create your views here.
 
 def index(request):
     return render(request, 'store/index.html')
 
 def bookDetailView(request, bid):
     template_name = 'store/book_detail.html'
-    context = {
-        'book': None, # set this to an instance of the required book
-        'num_available': None, # set this to the number of copies of the book available, or 0 if the book isn't available
-    }
-    # START YOUR CODE HERE
     
+    # List of available book id(s)
+    avail = []
+    for i in Book.objects.all():
+        avail.append(i.id)
     
+    # If the id in the request is in the list,  
+    if bid in avail:
+        book = Book.objects.filter(id = bid).first()
```

Comment:
This is not a good way to call ORM queries. This will run two queries on the database, one to append and the other to filter.